src/makeCalculator.js

```diff
@@ -3,8 +3,35 @@
 /**
  * @return {object}
  */
+
+// `add`, `subtract`, `multiply`, `divide`, `reset`, `operate`
+
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+    add(val) {
+      this.result += val;
+    },
+    subtract(val) {
+      this.result -= val;
+    },
+    multiply(val) {
+      this.result *= val;
+    },
+    divide(val) {
+      this.result /= val;
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    operate(operation, val) {
```

Comment:
The `operate` method can be improved by directly invoking the passed operation on `this.result` with the provided value instead of using `call`. This would simplify the implementation and avoid the unnecessary use of `this` binding, which is not needed as the methods `add`, `subtract`, `multiply`, `divide` are already using `this` to refer to the calculator object.

Reply:
[comment](https://github.com/mate-academy/js_advanced_calculator/pull/2333/files#r1683518982) not fixed